torch/_dynamo/variables/builder.py

```diff
@@ -744,29 +751,34 @@ def _clone_input(value):
     with preserve_rng_state():
         if example_value is None:
             example_value = get_fake_value(proxy.node, tx)
-        else:
-            # Note: Unfortunately, this can happen during tracing, and is valid enough for now to allow.
-            # TODO(voz): Find all the callsites and burn this down.
-            # Flipping it to an assert fails dozens of tests.
-            # TODO(ezyang): should attempt this burndown again
-            if not isinstance(example_value, torch._subclasses.FakeTensor):
-                # We shouldn't be doing this at all, see
-                # https://github.com/pytorch/torchdynamo/issues/1950
-                # But assuming we're doing it, the legacy behavior for
-                # subclasses was to perform a clone WITHOUT preserving
-                # the subclass.  It's not clear to me that's what you actually
-                # want, but whatever, I wouldn't have this cache at all.
-                with torch._C.DisableTorchFunction():
-                    proxy.tracer.real_value_cache[proxy.node] = _clone_input(
-                        example_value
-                    )
-                # NB: If we're ignoring subclass, then the expectation is you will
-                # take the returned TensorVariable and wrap it into a more
-                # accurate TensorVariable that is able to track subclass-ness;
-                # otherwise this is wrong!
-                example_value = wrap_to_fake_tensor_and_record(
-                    example_value, tx=tx, ignore_subclass=ignore_subclass
-                )
+
+        # Handle recursive calls here
+        elif isinstance(example_value, FakeTensor):
+            pass
+
+        elif isinstance(example_value, torch.Tensor):
+            # We shouldn't be doing this at all, see
+            # https://github.com/pytorch/torchdynamo/issues/1950
+            # But assuming we're doing it, the legacy behavior for
+            # subclasses was to perform a clone WITHOUT preserving
+            # the subclass.  It's not clear to me that's what you actually
+            # want, but whatever, I wouldn't have this cache at all.
+            with torch._C.DisableTorchFunction():
+                proxy.tracer.real_value_cache[proxy.node] = _clone_input(example_value)
+            # NB: If we're ignoring subclass, then the expectation is you will
+            # take the returned TensorVariable and wrap it into a more
+            # accurate TensorVariable that is able to track subclass-ness;
+            # otherwise this is wrong!
+            kwargs = {"ignore_subclass": ignore_subclass}
+            assert "source" in options
+            if options["source"] is None:
+                kwargs["static_shapes"] = True
+                kwargs["sname"] = "__constant_illegal_sname"
```

Comment:
is this ever checked anywhere? Why do we need a `__constant_illegal_sname` sname?